Kernel/semaphores.c

```diff
@@ -0,0 +1,163 @@
+#include <semaphores.h>
+#include <graphics.h>
+
+// A list might be implemented to track de asociated processes to a semaphore
+// Now just are being counted the number of processes, but this could lead to an inconsistency if,
+// for example, a process tries to close the same semaphore twice
+typedef struct {
+    Tsem semId; 
```

Comment:
(Just a tip) This seems to be only used with the resource namer. Another way to handle this is to not have `TSemaphore` store the id, and have the resource namer store the `semId` rather than the `TSemaphore*` (`void*` is just an integer after all)